pkgs/development/tools/mysql-shell/default.nix

```diff
@@ -0,0 +1,174 @@
+  '';
+in
+stdenv.mkDerivation rec{
+  pname = "mysql-shell";
+  version = "8.0.28";
+
+  srcs = [
+    (
+      fetchurl {
+        url = "https://cdn.mysql.com//Downloads/MySQL-Shell/mysql-shell-${version}-src.tar.gz";
+        sha256 = "sha256-xm2sepVgI0MPs25vu+BcRQeksaVhHcQlymreN1myu6c=";
+      }
+    )
+    (
+      fetchurl {
+        url = "https://dev.mysql.com/get/Downloads/MySQL-8.0/mysql-${version}.tar.gz";
+        sha256 = "sha256-2Gk2nrbeTyuy2407Mbe3OWjjVuX/xDVPS5ZlirHkiyI=";
+      }
+    )
+  ];
+
+  sourceRoot = "mysql-shell-${version}-src";
+
+  patchPhase = ''
+    runHook prePatch
+
+    patch ../mysql-${version}/cmake/fido2.cmake ${./fido2.cmake.patch}
+
+    substituteInPlace ../mysql-${version}/cmake/libutils.cmake --replace /usr/bin/libtool libtool
+    substituteInPlace ../mysql-${version}/cmake/os/Darwin.cmake --replace /usr/bin/libtool libtool
+    substituteInPlace cmake/libutils.cmake --replace /usr/bin/libtool libtool
+
+    runHook postPatch
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    git
+    bison
+  ] ++ lib.optionals (!stdenv.isDarwin) [ rpcsvc-proto ];
+
+  buildInputs = [
+    boost
+    curl
+    libedit
+    libssh
+    lz4
+    openssl
+    protobuf
+    readline
+    zlib
+    zstd
+    libevent
+    icu
+    re2
+    ncurses
+    libfido2
+    cyrus_sasl
+    openldap
+    v8
+  ] ++ pythonDeps ++ lib.optionals stdenv.isLinux [
+    numactl
+    libtirpc
+  ] ++ lib.optionals stdenv.isDarwin [
+    cctools
+    CoreServices
+    developer_cmds
+    DarwinTools
+  ];
+
+  dontConfigure = true;
+
+  preBuild = ''
+    cmake -DWITH_BOOST=system \
+      -DWITH_SYSTEM_LIBS=ON \
+      -DWITH_ROUTER=OFF \
+      -DWITH_UNIT_TESTS=OFF \
+      -DFORCE_UNSUPPORTED_COMPILER=1 \
+      -S ../mysql-${version} -B ../mysql-${version}-build
+
+    cmake --build ../mysql-${version}-build --parallel ''${NIX_BUILD_CORES:-1} --target mysqlclient mysqlxclient
+
+    export CXXFLAGS="-DV8_COMPRESS_POINTERS=1 -DV8_31BIT_SMIS_ON_64BIT_ARCH=1"
+
+    mkdir -p ../v8/lib
+    cp ${v8}/lib/libv8.a ../v8/lib/libv8_monolith.a
+
+    cmake -DMYSQL_SOURCE_DIR=../mysql-${version} \
+      -DMYSQL_BUILD_DIR=../mysql-${version}-build \
+      -DWITH_ZSTD=system \
+      -DWITH_LZ4=system \
+      -DWITH_ZLIB=system \
+      -DWITH_PROTOBUF=${protobuf} \
+      -DHAVE_V8=1 \
+      -DV8_INCLUDE_DIR=${v8}/include \
+      -DV8_LIB_DIR=../v8/lib \
+      -DHAVE_PYTHON=1 \
+      -S . -B ../mysql-shell-${version}-build
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    cmake --build ../mysql-shell-${version}-build --parallel ''${NIX_BUILD_CORES:-1}
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    cmake --install ../mysql-shell-${version}-build --prefix $out
+
+    echo '${site}' >> $out/lib/mysqlsh/python-packages/mysqlsh/__init__.py
+
+    runHook postInstall
+  '';
```

Comment:
Can't we use the default cmake phases?

Reply:
Before build MySQL Shell, We should build MySQL Server manually. I don't know how to do two cmake builds. So I choose to do this manually. Any suggestions?